tests/expected_tests.cc

```diff
@@ -0,0 +1,39 @@
+#include "gtest/gtest.h"
+#include "stout/expected.h"
+
+stout::expected<int, std::string> divide(int divident, int divider) {
+  if (!divider) {
+    return stout::make_unexpected("divide by zero");
+  } else {
+    return divident / divider;
+  }
+}
+
+TEST(Expected, SucceedResult) {
+  const auto result = divide(6, 3);
+
+  EXPECT_TRUE(result.has_value());
+  EXPECT_EQ(2, *result);
+}
+
+TEST(Expected, DivideByZero) {
+  const auto result = divide(6, 0);
+
+  EXPECT_TRUE(!result.has_value());
```

Comment:
similarly, will this work?
`EXPECT_EQ(result, stout::make_unexpected("divide by zero"));`

Reply:
It will work when I do the following:
`EXPECT_EQ(result, stout::make_unexpected(std::string{"divide by zero"}));`

Reply:
It's because type error is `const char* const &`:)

Reply:
Ooh, would this work as well? (in case you find this cleaner):

```c++
stout::make_unexpected<std::string>("divide by zero")
```